### PR TITLE
VideoCommon: add multithreaded asset loader and define a texture asset

### PR DESCRIPTION
**Pull request by iwubcode**

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
Needs a rebase.

File: Source/Core/VideoCommon/Assets/TextureAsset.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    ERROR_LOG_FMT(
+        VIDEO,
+        "Invalid custom texture size {}x{} for game texture asset '{}'. The aspect differs "
+        "from the native size {}x{}.",
+        first_mip.width, first_mip.height, GetAssetId(), native_width, native_height);
+    return false;
+  }
+
+  // Same deal if the custom texture isn't a multiple of the native size.
+  if (native_width != 0 && native_height != 0 &&
```
Comment (AdmiralCurtiss) [inline Source/Core/VideoCommon/Assets/TextureAsset.cpp:64]: Is a native_width or native_height of zero even valid? Maybe you want to catch that earlier and just return false for those cases.

Comment (iwubcode) [reply Source/Core/VideoCommon/Assets/TextureAsset.cpp:64]: It does seem odd, this was just copied from here:  https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/VideoCommon/HiresTextures.cpp#L329

**Comment by iwubcode** [comment]
@AdmiralCurtiss - thanks!  All good now assuming that comment isn't a blocker

**Pull request merged**